source/extensions/retry/host/omit_hosts/omit_hosts.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "envoy/upstream/retry.h"
+#include "envoy/upstream/upstream.h"
+
+#include "extensions/retry/host/well_known_names.h"
+
+namespace Envoy {
+class OmitHostsRetryPredicate : public Upstream::RetryHostPredicate {
+public:
+  OmitHostsRetryPredicate(envoy::api::v2::core::Metadata metadata_match_criteria,
+                          uint32_t retry_count)
+      : attempted_hosts_(retry_count), metadata_match_criteria_(metadata_match_criteria) {}
+
+  bool shouldSelectAnotherHost(const Upstream::Host& host) override {
+    const auto& match_criteria_filter_it = metadata_match_criteria_.filter_metadata().find(
```

Comment:
is it possible to reuse the existing metadata match code in https://github.com/envoyproxy/envoy/blob/master/source/common/router/metadatamatchcriteria_impl.h ?

Reply:
I'm not sure if metadatamatchcriteriaimpl would be useful here but I can reuse metadataLabelMatch() from https://github.com/envoyproxy/envoy/blob/master/source/common/config/metadata.h. Thanks for pointing this out!